WordPress/Classes/ViewRelated/Blaze Campaigns/BlazeCampaignsViewController.swift

```diff
@@ -234,10 +294,43 @@ extension BlazeCampaignsViewController: NoResultsViewControllerDelegate {
     }
 }
 
+// MARK: - Show / hide Promote button
+
+extension BlazeCampaignsViewController: Subscriber {
+
+    typealias Input = Bool
+    typealias Failure = Never
+
+    func receive(subscription: Subscription) {
+        subscription.request(.unlimited)
+    }
+
+    func receive(_ input: Bool) -> Subscribers.Demand {
```

Comment:
Please, use [`https://developer.apple.com/documentation/combine/fail/sink(receivevalue:)`].

```swift
scrollViewTranslationPublisher.sink { [weak self] value in
    // ...
}.store(in: &cancellables)
```

`Subscriber` is a low-level API.

> Combine provides the following subscribers as operators on the [Publisher](https://developer.apple.com/documentation/combine/publisher) type:
> - [`sink(receiveCompletion:receiveValue:)`](https://developer.apple.com/documentation/combine/publisher/sink(receivecompletion:receivevalue:)) executes arbitrary closures when it receives a completion signal and each time it receives a new element.
> - [`assign(to:on:)`](https://developer.apple.com/documentation/combine/publisher/assign(to:on:)) writes each newly-received value to a property identified by a key path on a given instance.

Reply:
On second thought, does it need to be a Publisher? I would suggest calling it directly from `scrollViewDidScroll`.

Reply:
> I would suggest calling it directly from scrollViewDidScroll.

Fixed! [c65bc8b](https://github.com/wordpress-mobile/WordPress-iOS/pull/21022/commits/c65bc8b4631fb83d847b76b4224ea8e208c12bac)